src/cargo/sources/git/oxide.rs

```diff
@@ -186,27 +186,102 @@ fn amend_authentication_hints(
     }
 }
 
+/// The reason we are opening a git repository.
+///
+/// This can affect the way we open it and the cost associated with it.
+pub enum OpenMode {
+    /// We need `git_binary` configuration as well for being able to see credential helpers
+    /// that are configured with the `git` installation itself.
+    /// However, this is slow on windows (~150ms) and most people won't need it as they use the
+    /// standard index which won't ever need authentication, so we only enable this when needed.
+    ForFetch,
+}
+
 /// Produce a repository with everything pre-configured according to `config`. Most notably this includes
-/// transport configuration.
-pub fn open_repo(repo_path: &std::path::Path, config: &Config) -> CargoResult<git::Repository> {
+/// transport configuration. Knowing its `purpose` helps to optimize the way we open the repository.
+pub fn open_repo(
+    repo_path: &std::path::Path,
+    config: &Config,
+    purpose: OpenMode,
+) -> CargoResult<git::Repository> {
     git::open_opts(repo_path, {
         let mut opts = git::open::Options::default();
-        let timeout = HttpTimeout::new(config)?;
-
-        // We need `git_binary` configuration as well for being able to see credential helpers
-        // that are configured with the `git` installation itself.
-        // However, this is slow on windows (~150ms) and most people won't need it as they use the
-        // standard index which won't ever need authentication.
-        // TODO: This is certainly something to make configurable, at the very least on windows.
-        //       Maybe it's also something that could be cached, all we need is the path to the configuration file
-        //       which usually doesn't change unless the installation changes. Maybe something keyed by the location of the
-        //       binary along with its fingerprint.
         opts.permissions.config = git::permissions::Config::all();
-        opts.config_overrides([
-            format!("gitoxide.http.connectTimeout={}", timeout.dur.as_millis()),
-            format!("http.lowSpeedLimit={}", timeout.low_speed_limit),
-            format!("http.lowSpeedTime={}", timeout.dur.as_secs()),
-        ])
+        opts.permissions.config.git_binary = matches!(purpose, OpenMode::ForFetch);
```

Comment:
The problem with `matches!` is we will miss re-evaluating this when adding new variants

Reply:
A good point! It's fixed here: ce08e8b3deb67706f998c70714764851504967f1.